### DISCUSSION TITLE
changes in WLAN module and hostname

### DISCUSSION THREAD

**Discussion by dusan-zervan** [micropython/micropython#11046] [General]
I have found differences between 1.19.1 and latest nightly builds in WLAN module: [documentation](https://docs.micropython.org/en/latest/library/network.WLAN.html)
![MicroPython 1 20 WLAN](https://user-images.githubusercontent.com/36242403/225611256-b80642ae-d421-429a-a954-7a83320851bc.png)
It ruins my old programs, but I can adapt. However, why the new function `network.hostname()` is not accepting hostname longer than 15 characters? Is it a mistake or intended? I usually use longer hostnames.

**Comment by scruss** [comment]
15 characters is the old NetBIOS limit, but that might be coincidental.

It seems deliberate, from the recent changes to [extmod/modnetwork.h](https://github.com/micropython/micropython/commit/a3773026238c7c779cad79aa9d9c468e680e7004#diff-3eb4b8f1e0b2ae9c3b23247caad0f60c39ff4fa040fc9a68bb4e168ec2627249)
